### ISSUE TITLE
MA0134 not triggered by synchronous Task-returning local method in async method

### ISSUE THREAD

**Issue by jackalvrus** [meziantou/Meziantou.Analyzer#597]
### Version of the Meziantou.Analyzer NuGet package

2.0.84

### Rule Identifier

MA0134

### Target Framework

net6.0

### C# Language version

C# 10

### Description

In our production code, I ran into a scenario where a synchronous task-returning but non-async local method inside an async method is not detected by MA0134. Adding a test confirmed that it was not being detected. 

### Reproduction Steps

This test fails:

```c#
    [Fact]
    public async Task Report_TaskInLocalTaskReturningFunction()
    {
        await CreateProjectBuilder()
            .WithSourceCode("""
                using System.Threading.Tasks;
                class Test
                {
                    async Task A()
                    {
                        await B();

                        Task B()
                        {
                            [||]Task.Delay(0);
                            return Task.CompletedTask;
                        }
                    }
                }
                """)
            .ValidateAsync();
    }
```

### Other information

Simply removing the early break-out for async methods in the lambda passed to `RegisterSymbolStartAction` seems to fix the issue.
